### ISSUE TITLE
[FKA] Keyboard focus resets to the top of the list after navigating back

### ISSUE THREAD

**Issue by preethi0228-git** [flutter/flutter#192182]
### Steps to reproduce

1. Clone the a11y assessments app from flutter repo and build the app in iPad/iPhone
2. Enable Full Keyboard Access on the iPad.
3. Connect a physical keyboard.
4. Launch the application.
5. Navigate through the list of interactive elements using the Tab key.
6. Move focus to an element in the list, for example AutoComplete.
7. Press Space to activate/select the focused element.
8. Navigate back to the previous screen.
9. Observe where the keyboard focus is placed.

### Expected results

When returning to the previous screen, keyboard focus should be restored to the element that had focus before navigation.

For example, if AutoComplete had focus before navigating away, focus should return to AutoComplete when the user comes back.

### Actual results

When navigating back, the keyboard focus is reset to the first/top element in the list (AboutListTile) instead of returning to the previously focused element.

**Note**

- VoiceOver:  Focus remains at the same position when returning. 
- Full Keyboard Access / Physical Keyboard:  Focus moves to the top of the screen.

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

 
https://github.com/user-attachments/assets/cc6efe63-17a1-43d4-bd5b-417941396bae

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.48.0-1.0.pre-524, on macOS 27.0 26A5421a darwin-arm64, locale en-US) [1,364ms]
    • Flutter version 3.48.0-1.0.pre-524 on channel master at /Users/preethikandukuri/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f5f6313e8c (4 hours ago), 2026-08-31 05:22:09 +0000
    • Engine revision f5f6313e8c
    • Dart version 3.14.0 (build 3.14.0-180.0.dev)
    • DevTools version 2.61.0-dev.0
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android,
      enable-ios, cli-animations, enable-native-assets, enable-record-use, enable-swift-package-manager,
      omit-legacy-version-file, enable-lldb-debugging, enable-uiscene-migration, enable-riscv64, enable-hcpp

[✓] Android toolchain - develop for Android devices (Android SDK version 37.0.0) [455ms]
    • Android SDK at /Users/preethikandukuri/Library/Android/sdk
    • Emulator version 37.1.11.0 (build_id 15917651) (CL:N/A)
    • Platform android-37.0, build-tools 37.0.0
    • ANDROID_HOME = /Users/preethikandukuri/Library/Android/sdk
    • Java binary at: /Applications/Android Studio RC.app/Contents/jbr/Contents/Home/bin/java
      This JDK is specified in your Flutter configuration.
      To change the current JDK, run: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 25.0.3+-15898627-b508.16)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 27.0) [657ms]
    • Xcode at /Applications/Xcode-beta276.app/Contents/Developer
    • Build 27A5252f
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [4ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Connected device (5 available) [6.0s]
    • sdk gphone16k arm64 (mobile) • emulator-5554                        • android-arm64  • Android 17 (API 37)
      (emulator)
    • iPad (wireless) (mobile)     • 00008122-001A08DC22B9801C            • ios            • iOS 27.0 24A5424a
    • iPhone 17 Pro (mobile)       • 6A22C9A6-0DD2-4ADA-84BB-ECB0DDE39E6C • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-27-0 (simulator)
    • macOS (desktop)              • macos                                • darwin-arm64   • macOS 27.0 26A5421a
      darwin-arm64
    • Chrome (web)                 • chrome                               • web-javascript • Google Chrome 152.0.7977.65

[✓] Network resources [893ms]
    • All expected network resources are available.

```

</details>

**Comment by hannah-hyj** [comment]
routing to iOS team.

**Comment by LongCatIsLooong** [comment]
@preethi0228-git could you try this in a UIKit app and see if the keyboard focus resets?

**Comment by preethi0228-git** [comment]
Hi @LongCatIsLooong ,

Verified the same scenario in the native UIKit app and Settings app with Full Keyboard Access enabled and the same physical keyboard. In both cases, focus was restored to the previously focused element after navigating back. The focus restoration issue was observed only in the Flutter app. 

Attaching the videos for the same.

Thanks.

https://github.com/user-attachments/assets/6e3669a5-84d9-4490-b8fc-6ee6da2aacd0


https://github.com/user-attachments/assets/037158dd-bd0c-4e35-9e17-484d3d756d14
